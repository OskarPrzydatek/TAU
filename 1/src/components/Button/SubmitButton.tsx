type SubmitButtonProps = {
  label: string;
  dataTestId?: string;
};

export default function SubmitButton({ label, dataTestId }: SubmitButtonProps) {
  return (
    <button data-testid={dataTestId} type="submit">
      {label}
    </button>
  );
}
