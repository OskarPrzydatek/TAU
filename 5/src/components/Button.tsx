type ButtonProps = {
  label: string;
  dataTestID?: string;
  onClick: () => void;
};

export default function Button({ label, dataTestID, onClick }: ButtonProps) {
  return (
    <button data-testid={dataTestID} onClick={onClick}>
      {label}
    </button>
  );
}
