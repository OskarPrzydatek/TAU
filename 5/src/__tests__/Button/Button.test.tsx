import { fireEvent, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Button from '../../components/Button';

const mockOnClick = jest.fn();

describe('Button', () => {
  test('Component Snapshot', () => {
    const tree = renderer
      .create(<Button label="Add" onClick={mockOnClick} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('ensure button works correctly', () => {
    render(
      <Button dataTestID="add-button" label="Add" onClick={mockOnClick} />
    );
    const button = screen.getByTestId('add-button');
    expect(button).toHaveTextContent('Add');
    fireEvent.click(button);
    expect(mockOnClick).toHaveBeenCalled();
  });
});
