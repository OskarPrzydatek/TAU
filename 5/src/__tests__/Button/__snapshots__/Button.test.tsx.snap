// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Snapshot 1`] = `
<button
  onClick={[MockFunction]}
>
  Add
</button>
`;
